dblookupext/esdtSupply/esdtSuppliesProcessor_test.go

```diff
@@ -71,28 +73,186 @@ func TestProcessLogsSaveSupply(t *testing.T) {
+			if string(key) == "processed-block" {
+				return nil
+			}
+
+			supplyKey := string(token) + "-" + hex.EncodeToString(big.NewInt(2).Bytes())
 			require.Equal(t, supplyKey, string(key))
 
 			var supplyESDT SupplyESDT
 			_ = marshalizer.Unmarshal(&supplyESDT, data)
 			require.Equal(t, big.NewInt(30), supplyESDT.Supply)
 
+			wasPutCalled = true
+
 			return nil
 		},
 	}
 
 	suppliesProc, err := NewSuppliesProcessor(marshalizer, suppliesStorer, &storageStubs.StorerStub{})
 	require.Nil(t, err)
 
-	err = suppliesProc.ProcessLogs(0, logs)
+	err = suppliesProc.ProcessLogs(6, logs)
 	require.Nil(t, err)
+
+	require.True(t, wasPutCalled)
+}
+
+func TestProcessLogsSaveSupplyShouldUpdateSupplyMintedAndBurned(t *testing.T) {
+	t.Parallel()
+
+	token := []byte("nft-0001")
+	logsCreate := []*data.LogData{
+		{
+			TxHash: "txLog",
+			LogHandler: &transaction.Log{
+				Events: []*transaction.Event{
+					{
+						Identifier: []byte("something"),
+					},
+					{
+						Identifier: []byte(core.BuiltInFunctionESDTNFTCreate),
+						Topics: [][]byte{
+							token, big.NewInt(2).Bytes(), big.NewInt(10).Bytes(),
+						},
+					},
+				},
+			},
+		},
+		{
+			TxHash: "log",
+		},
+	}
+	logsAddQuantity := []*data.LogData{
+		{
+			TxHash: "txLog",
+			LogHandler: &transaction.Log{
+				Events: []*transaction.Event{
+					{
+						Identifier: []byte("something"),
+					},
+					{
+						Identifier: []byte(core.BuiltInFunctionESDTNFTAddQuantity),
+						Topics: [][]byte{
+							token, big.NewInt(2).Bytes(), big.NewInt(50).Bytes(),
+						},
+					},
+				},
+			},
+		},
+		{
+			TxHash: "log",
+		},
+	}
+
+	logsBurn := []*data.LogData{
+		{
+			TxHash: "txLog",
+			LogHandler: &transaction.Log{
+				Events: []*transaction.Event{
+					{
+						Identifier: []byte("something"),
+					},
+					{
+						Identifier: []byte(core.BuiltInFunctionESDTNFTBurn),
+						Topics: [][]byte{
+							token, big.NewInt(2).Bytes(), big.NewInt(30).Bytes(),
+						},
+					},
+				},
+			},
+		},
+		{
+			TxHash: "log",
+		},
+	}
+
+	membDB := testscommon.NewMemDbMock()
+	marshalizer := testscommon.MarshalizerMock{}
+	numTimesCalled := 0
+	suppliesStorer := &storageStubs.StorerStub{
+		GetCalled: func(key []byte) ([]byte, error) {
+			if string(key) == "processed-block" {
+				pbn := ProcessedBlockNonce{Nonce: 5}
+				pbnB, _ := marshalizer.Marshal(pbn)
+				return pbnB, nil
+			}
+			supplyKey := string(token) + "-" + hex.EncodeToString(big.NewInt(2).Bytes())
+			if string(key) == supplyKey {
+				val, err := membDB.Get(key)
+				if err != nil {
+					return nil, storage.ErrKeyNotFound
+				}
+				return val, nil
+			}
+			return nil, storage.ErrKeyNotFound
+		},
+		PutCalled: func(key, data []byte) error {
+			supplyKey := string(token) + "-" + hex.EncodeToString(big.NewInt(2).Bytes())
+			if string(key) == supplyKey {
+				switch numTimesCalled {
+				case 0:
+					supplyEsdt := getSupplyESDT(marshalizer, data)
+					require.Equal(t, big.NewInt(10), supplyEsdt.Supply)
```

Comment:
I would have extracted these into constants `providedCreateValue`, `providedBurnValue` and then create some `expectedAfterAdd`, `expectedAfterBurn`. Same for previous test

Reply:
extracted some constants